src/perf.cc

```diff
@@ -163,12 +174,17 @@ int perfdisown(void *region, size_t size) {
 
 // return attr sorted by priority
 std::vector<perf_event_attr>
-all_perf_configs_from_watcher(const PerfWatcher *watcher, bool extras) {
+all_perf_configs_from_watcher(const PerfWatcher *watcher, bool extras,
+                              PerfClockSource perf_clock_source) {
   std::vector<perf_event_attr> ret_attr;
-  ret_attr.push_back(perf_config_from_watcher(watcher, extras));
+  auto attr = perf_config_from_watcher(watcher, extras);
```

Comment:
Minor: It would have been OK to add a parameter to this function to get the perf_clock_source part of object creation.

Reply:
Done